File: 2022/3/go/solution2.go
package main

import (
	"fmt"
	"io/ioutil"
	// "strconv"
	"strings"
)

func main() {
	filebuffer, err := ioutil.ReadFile("../input.txt")
	if err != nil {
		return
	}
	inputdata := strings.Split(string(filebuffer), "\r\n")

	total_sum := 0

	for i := 0; i < len(inputdata); i += 3 {
		first_rucksack := inputdata[i]
		second_rucksack := inputdata[i+1]
		third_rucksack := inputdata[i+2]
		m := make(map[rune]rune)
		for _, c := range first_rucksack + second_rucksack + third_rucksack {
			var value rune
			if c >= 96 {
				value = c - (96)
			} else {
				value = c - 65 + 27
			}
			m[c] = value
		}
		for k, v := range m {
			if strings.Contains(first_rucksack, string(k)) && strings.Contains(second_rucksack, string(k)) && strings.Contains(third_rucksack, string(k)) {
				fmt.Printf("char=%1c rune=%3v priority=%2v ", k, k, v)
				total_sum += int(v)
				break
			}
		}
		fmt.Printf("%5v %25v %25v %25v %v\n", total_sum, first_rucksack, second_rucksack, third_rucksack, m)
	}
	fmt.Println(total_sum)
}

File: 2019/2/python/solution2.py
total=0
for noun in range(100):
	for verb in range(100):
		with open("../input.txt") as f:
			array = list(map(int, f.read().split(',')))
			array[1], array[2] = noun, verb
			# print(array, int(len(array)/4))
			for x in range(0, len(array), 4):
				# print(array[x:x+4])
				if array[x]==1:
					array[array[x+3]] = array[array[x+1]] + array[array[x+2]]	
				elif array[x]==2:
					array[array[x+3]] = array[array[x+1]] * array[array[x+2]]	
				elif array[x]==99:
					break

			if array[0]==19690720:
				# print(noun, verb, array[0], array[0]==19690720)
				break
		if array[0]==19690720:
			break		
	if array[0]==19690720:
		break

print(100 * noun + verb)

# 19690720, 3146